src/intervals/reduction.jl

```diff
@@ -0,0 +1,24 @@
+# This file is part of the IntervalArithmetic.jl package; MIT licensed
+
+"""
+The file contains the reduction functions stated in the section 12.12.12 with correctly rounded results.
+r is the rounding direction which takes 0.0 for RoundingToZero , 0.5 for RoundingNearest , +Inf for RoundingUp and -Inf for RoundingDown.
+"""
+function vector_sum(v :: Vector{T} , r :: RoundingMode) where T
+    sum1 = sum(BigFloat.(v))
```

Comment:
This sum might introduce rounding errors? How do you know which precision to use?

Reply:
Same for all the other functions.

Reply:
According to the standard, for reduction functions, `evaluation is as if in exact arithmetic up to the ﬁnal rounding, with no possibility of intermediate overﬂow or underﬂow`. Do we have an alternative of `sum` for `BigFloats`?

Reply:
I think the only way is to do the loop ourselves, either explicitely or using `reduce`, and using the rounded `+` e.g.

```julia
sum(v::Vector, r::RoundingMode) = reduce((x, y) -> +(x, y, r), v)
```

Reply:
I believe these functions are supposed to use long accumulators, i.e. with high precision.